parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java

```diff
@@ -192,7 +198,8 @@ private CompressionCodecName codec() {
             .build();
 
         return new org.apache.iceberg.parquet.ParquetWriter<>(
-            conf, file, schema, rowGroupSize, metadata, createWriterFunc, codec(), parquetProperties);
+            conf, file, schema, rowGroupSize, statsTruncateLength, metadata,
+                createWriterFunc, codec(), parquetProperties);
```

Comment:
The `else` case just below is where `statsTruncateLength` needs to be passed to `ParquetWriteAdapter`.